tests/7.AlgoSDK.js

```diff
@@ -33,6 +33,45 @@ describe('Algosdk (AKA end to end)', function () {
             let o = "Hi there";
             assert.deepStrictEqual(o, algosdk.decodeObj(algosdk.encodeObj(o)));
         });
+
+        it('should not mutate unsigned transaction when going to or from encoded buffer', function () {
+            let to = "PNWOET7LLOWMBMLE4KOCELCX6X3D3Q4H2Q4QJASYIEOF7YIPPQBG3YQ5YI";
+            let fee = 4;
+            let amount = 1000;
+            let firstRound = 12466;
+            let lastRound = 13466;
+            let genesisID = "devnet-v33.0";
+            let genesisHash = "JgsgCaCTqIaLeVhyL6XlRu3n7Rfk2FxMeK+wRSaQ7dI=";
+            let closeRemainderTo = "IDUTJEUIEVSMXTU4LGTJWZ2UE2E6TIODUKU6UW3FU3UKIQQ77RLUBBBFLA";
+            let note = new Uint8Array(Buffer.from("6gAVR0Nsv5Y=", "base64"))
+            let from = to;
+            let txnAsObj = algosdk.makePaymentTxn(from, to, fee, amount, closeRemainderTo, firstRound, lastRound, note, genesisHash, genesisID);
+            let txnAsBuffer = algosdk.unsignedTransactionToBuffer(txnAsObj);
+            let txnAsObjRecovered = algosdk.bufferToUnsignedTransaction(txnAsBuffer);
+            let txnAsBufferRecovered = algosdk.unsignedTransactionToBuffer(txnAsObjRecovered);
+            assert.deepStrictEqual(txnAsBuffer, txnAsBufferRecovered);
```

Comment:
Golden being added. `txnAsObj` won't successfully compare as equal to `txnAsObjRecovered`, because fields that are absent in `txnAsObj` get reencoded as being present in `txnAsObjRecovered`, but having a value of `undefined`.